src/Workspaces/Remote/ServiceHub/Services/DesignerAttribute/RemoteDesignerAttributeIncrementalAnalyzer.cs

```diff
@@ -74,121 +90,82 @@ private async Task AnalyzeProjectAsync(Project project, Document? specificDocume
+                // If we don't have a path for this document, we cant proceed with it.
+                // We need that path to inform the project system which file we're referring to.
+                if (document.FilePath == null)
+                    continue;
+
+                // If nothing has changed at the top level between the last time we analyzed this document and now, then
+                // no need to analyze again.
+                if (_documentToLastReportedInformation.TryGetValue(document.Id, out var existingInfo) &&
+                    existingInfo.projectVersion == projectVersion)
+                {
+                    continue;
+                }
+
+                tasks.Add(ComputeDesignerAttributeDataAsync(designerCategoryType, document, cancellationToken));
             }
 
             return await Task.WhenAll(tasks).ConfigureAwait(false);
         }
 
-        private static async Task<(Document, DesignerAttributeData?, bool changed)> ComputeDesignerAttributeDataAsync(
-            IPersistentStorage storage, VersionStamp projectVersion, INamedTypeSymbol? designerCategoryType,
-            Document document, CancellationToken cancellationToken)
+        private static async Task<(Document document, DesignerAttributeData data)> ComputeDesignerAttributeDataAsync(
+            INamedTypeSymbol? designerCategoryType, Document document, CancellationToken cancellationToken)
         {
             try
             {
-                // If we don't have a path for this document, we cant proceed with it.
-                // We need that path to inform the project system which file we're referring to.
-                if (document.FilePath == null)
-                    return default;
-
-                // First check and see if we have stored information for this doc and if that
-                // information is up to date.
-                using var stream = await storage.ReadStreamAsync(document, DataKey, cancellationToken).ConfigureAwait(false);
-                using var reader = ObjectReader.TryGetReader(stream, cancellationToken: cancellationToken);
-                var persisted = TryReadPersistedInfo(reader);
-                if (persisted.category != null && persisted.projectVersion == projectVersion)
-                {
-                    // We were able to read out the old data, and it matches our current project
-                    // version.  Just return back that nothing changed here.  We won't tell VS about
-                    // this, and we won't re-persist this later.
-                    return default;
-                }
+                Contract.ThrowIfNull(document.FilePath);
 
                 // We either haven't computed the designer info, or our data was out of date.  We need
                 // So recompute here.  Figure out what the current category is, and if that's different
                 // from what we previously stored.
```

Comment:
Comment seems stale -- should be deleted now?

Reply:
it's still accurate.  we have to comptue this and compare against whatever we last reported.  teh difference now is that we keep track of what we reported in memory instead of on disk.

Reply:
Well, it's still correct, but previously that function was doing the storage comparison. Now this function is blissfully ignorant of how it's being used -- does the comment move elsewhere then?

Reply:
i'll revise in next pr.

Reply:
revised in next pr.